projects/06/cmd/assemble/utils.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+)
+
+// parse the argument of filename and create filepath variable
+func parseArgs(args ...string) (string, error) {
+	flg := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	f := flg.String("filename", "add/Add", "filename flag")
+	flg.Parse(args)
+	if *f == "" {
+		return "", fmt.Errorf("error: filename is empty")
+	}
+	return fmt.Sprintf("test/%s", *f), nil
+}
+
+// scan the content in filepath and convert to binary string each line
+func scanline(fp *os.File) ([]string, error) {
+	scanner := bufio.NewScanner(fp)
+	lines := make([]string, 0)
+	count := 0
+	preDict()
+	for scanner.Scan() {
```

Comment:
I think you should indicate this for loop is the first scan, and for creating a label table.